contract/src/orca.contract.js

```diff
@@ -1,111 +1,287 @@
+    marshaller,
+    storageNode,
+    timer,
+    localchain,
+    agoricNames,
+  } = privateArgs;
+  trace('orchestration: ', orchestration);
+  trace('marshaller: ', marshaller);
+  trace('storageNode: ', storageNode);
+  trace('timer: ', timer);
+  trace('localchain: ', localchain);
+  trace('agoricNames: ', agoricNames);
+  const { orchestrate, zone, vowTools, zoeTools, makeRecorderKit, asyncFlowTools} = provideOrchestration(
+    zcf,
+    baggage,
+    privateArgs,
+    privateArgs.marshaller,
+  );
+
+  const { asVow, watch } = vowTools;
+  trace('orchestrate: ', orchestrate);
+  trace('zone: ', zone);
+  trace('vowTools: ', vowTools);
+  trace('asVow: ', asVow);
+  trace('watch: ', watch);
+  trace('zoeTools: ', zoeTools);
+  trace('makeRecorderKit: ', makeRecorderKit);
+  trace('asyncFlowTools: ', asyncFlowTools);
+
+  /**
+   * @param {Zone} zone
+   * @param {{ zcf: ZCF; vowTools: VowTools }} io
+   */
+  const wrapper = (zone, { zcf, vowTools }) => {
+    const transfer = vowTools.retriable(
+      zone,
+      'transfer',
+      /**
+       * @type {transfer}
+       */
+      async (srcSeat, localAccount, give, amt, localAddress, remoteAddress) => {
+        !srcSeat.hasExited() || Fail`The seat cannot have exited.`;
+        const { zcfSeat: tempSeat, userSeat: userSeatP } = zcf.makeEmptySeatKit();
+        trace('tempSeat:', tempSeat);
+        const userSeat = await userSeatP;
+        trace('userSeat:', userSeat);
+        atomicTransfer(zcf, srcSeat, tempSeat, give);
+        tempSeat.exit();
+        const promises = Object.entries(give).map(async ([kw, _amount]) => {
+          trace("kw::", kw)
+          trace("_amount", _amount)
+          trace("amt", amt)
+        });
+        const pmt = await E(userSeat).getPayout("Deposit");
+        trace("pmt:", pmt)
+        trace("amt:", amt)
         
+        const watcher = zone.exo(
+          `watcher-transfer-${localAddress.value}-to-${remoteAddress.value}`, // Error: key (a string) has already been used in this zone and incarnation -- perhaps use timestamp or offerid as well?
+           M.interface('watcher for transfer', {
+              onFulfilled: M.call(M.any()).optional(M.any()).returns(VowShape),
+            }
+          ),
+          {
+            /**
+             * @param {any} _result
+             * @param {bigint} value
+             */
+            onFulfilled(
+              _result, 
+              value
+            ) {
+              trace("inside onFulfilled:", value)
+              return watch(localAccount.transfer(
+                {
+                  denom: "ubld",
+                  value: value/2n,
+                },
+                remoteAddress
+              ))                    
+            },
+          },
+        );
+
+        trace("about to watch transfer, watcher v0.16")
+        trace("watcher", watcher)
+        watch(
+          E(localAccount).deposit(pmt),
+          watcher,
+          BigInt(amt.value),
+        );
```

Comment:
okay that makes sense. I'm just dividing by two as a demo/debug to show two accounts being funded, because if we transfer the entire amount in the end, sometimes it's not easily seen that localaccount ever had funds on the frontend. Just a debug thing for now

Reply:
going to try this suggestion without watcher

Reply:
Ah, I see what you mean here, its inside of `vowTools.retriable`, not just the handler! good point

Reply:
still encountering this when trying to do that, opened this ticket to track it @0xpatrickdev :
https://github.com/Agoric/dapp-orchestration-basics/issues/13#issue-2464084230